### PR TITLE
stop github actions from executing specific tasks on forks

### PR DESCRIPTION
**Pull request by fbegyn**
Currently github actions keep sending me mails about things that should only run on the prometheus organisation actions. This change makes sure to check who owns the repository before running the CI workflow.

I'm not fully convinced that this is the correct action to take, but it can function as a discussion starter.

@roidelapluie

### CODE REVIEW

File: .github/workflows/buf.yml [hunk 1]
```diff
@@ -7,6 +7,7 @@ jobs:
   buf:
     name: lint and publish
     runs-on: ubuntu-latest
+    if: github.repository_owner == 'prometheus'
```
Comment (roidelapluie) [inline .github/workflows/buf.yml:10]: it seems like buf should work everywhere. Maybe move the if at the push action?

**Comment by roidelapluie** [comment]
Is there an attribute that could run it if the repo is not a fork instead of hardcoding prometheus?

**Comment by fbegyn** [comment]
@roidelapluie to my knowledge there is no attribute that would allow us to differentiate like that. There is [a feature request open on github action](https://github.com/orgs/community/discussions/9098) for something that would allow ignoring forks, but even there the initial idea seems to be hardcoding organisations/repos.

**Comment by fbegyn** [comment]
So after looking over some further documentation, it might be possible to insert a check based on `gh repo view $(git config --get remote.origin.url) --json isFork` to the CI workflow, which would cause it to only run when not on a fork.

**Comment by fbegyn** [comment]
Some tests seem to fail, but the 2 step process and checking the output result of gh CLI seems to work. Only question is … does everything have the “remote.origin.url”? 🤔

**Comment by roidelapluie** [comment]
I am convinced we could hardcode     if: github.repository_owner == 'prometheus' for now.

**Comment by fbegyn** [comment]
rebased on `main` and switched to the hard-coded community check

**Comment by roidelapluie** [comment]
Thanks!

**Pull request merged**